New GUI

tuxinal opened the pull request:
Implements the new GUI proposed in #28 

![image](https://user-images.githubusercontent.com/24763016/197781676-ed631897-ad88-4436-a3b6-337d61f53ab8.png)

I haven't tested this on Windows and other python versions besides 3.10 yet. please do so if you have the chance
_(you might also have to install the [Inter](https://rsms.me/inter/) font until i figure out how to include it in the application itself)_

tuxinal committed:
New GUI

tuxinal commented:
@nsde i have made some changes to how the `get_dir`/`set_dir` functions work in `main.py`. the current implementation made the packwiz bootstrapper appear inside the home directory on linux systems and it seems unnecessary as far as i can tell. i have discussed this with osfanbuff on discord and they said it should be okay. i wanted to ask if you think there is any reason to keep the current functionality.

nsde commented:
LGTM, but:
- [Accessing the API may be problematic because it can lead to ratelimiting](https://github.com/Fabulously-Optimized/vanilla-installer/compare/main...tuxinal:vanilla-installer:main#diff-253c6a5418c0bf267e44dfb8f97def5223ece1268f602d15ca8813b76313d410R283)
- The theme dictionary keys have in-descriptive names, such as `subtext1` and `subtext0`

Also, as of right now, the fork don't actually implement the new GUI, but the commits look like preparations for that.

osfanbuff63 commented:
> Also, as of right now, the fork don't actually implement the new GUI, but the commits look like preparations for that

Tested it briefly locally, and it does seems to implement it. Haven't gone through everything yet, so I'll leave a review once I get through it.

Only thing I noticed is that it doesn't work on 3.11 but that's pyside6's fault, we'll just need to wait there.

tuxinal commented:
the theme names are based on [catppuccin](https://github.com/catppuccin/catppuccin)'s names for the colors. i suppose i can change them to be more descriptive.
what method would you suggest to avoid rate limits?

osfanbuff63 commented on vanilla-installer/main.py:
```diff
@@ -318,10 +273,28 @@ def create_profile(mc_dir: str, version_id: str) -> None:
+    return_value = []
+    try:
+        response = requests.get(
+            "https://api.github.com/repos/Fabulously-Optimized/fabulously-optimized/contents/Packwiz").json()
```
I think probably the easiest way to combat ratelimiting (which I believe is IP-based when unauthenticated, so it *shouldn't* cause other users trouble) is probably to cache the list and not make the request if the cache is found - cache basically meaning opening a text file and storing it there, although there's surely a better way to do that.

tuxinal replied:
We can also add a settings menu with an option to add a github api key, i doubt anyone who's gonna install FO more than 60 times in an hour doesn't have a github account

osfanbuff63 replied:
If you or anyone else wants to do a separate PR for this that's fine, but I'd kinda like to get this merged 😅

osfanbuff63 commented on vanilla-installer/theme.py:
```diff
@@ -40,37 +40,48 @@ def is_dark(to_dark: bool = None) -> bool:
         return open(FILE, "w", encoding="utf-8").write("dark")
     return open(FILE, encoding="utf-8").read() == "dark"
 
+# colors from catppuccin latte and mocha https://github.com/catppuccin/catppuccin
```
Probably worth adding a mention in credits.md for this one. Speaking of credits, is it worth adding a button for that?

tuxinal replied:
The icons are also from feathericons. we can add that to credit.md too

osfanbuff63 commented on vanilla-installer/gui.py:
```diff
@@ -22,185 +35,234 @@
+class Ui_MainWindow(object):
+    def setupUi(self, MainWindow: QMainWindow):
+        if not MainWindow.objectName():
+            MainWindow.setObjectName("MainWindow")
```
Seems at least on Windows, the window name is taken from here - so should probably be `VanillaInstaller` or something similar.

tuxinal replied:
can you try the new commit and see if the title changed?

osfanbuff63 replied:
Once I'm home I'll try it yeah

osfanbuff63 replied:
Yep, works now

osfanbuff63 reviewed (commented):
As mentioned earlier, Qt doesnt have a build out yet for python 3.11 (although likely all that's needed is a dep update) which isn't your fault ofc - so it doesn't work on 3.11 yet.

A couple bugs here and there, but overall it's pretty good (and works)! It might be worth adding a `QIcon`, and it doesn't scale when you maximize it, but those are relatively trivial.

tuxinal committed:
Suggestions
* change color names
* add credits
* change window title

osfanbuff63 commented on vanilla-installer/gui.py:
```diff
@@ -1,13 +1,23 @@
+import minecraft_launcher_lib as mll
 
 # External
 import requests
```
```suggestion

```
unused import, not a huge deal though
(same with logger but I can't leave a review comment there)

osfanbuff63 commented on vanilla-installer/gui.py:
```diff
@@ -22,185 +32,234 @@
 parser.add_argument("--safegui", type=bool)
 parser.add_argument("--litegui", type=bool)
```
These are unused now, the GUI isn't meant to be invoked via a CLI anyway since we have a dedicated CLI.

osfanbuff63 reviewed (commented):
Couple things, but otherwise looking pretty good. If you (or anyone else) wants to do a PR for a settings menu for an API token or something like that, feel free - but I'd kinda like to get this merged if possible.

tuxinal committed:
Add settings menu
currently contains options to change the gh api key to be used
also make sure the installation process is only ran once at a time

osfanbuff63 commented on vanilla-installer/main.py:
```diff
@@ -81,14 +65,48 @@ def get_dir() -> str:
+def set_gh_auth(user: str, key: str) -> bool | None:
+    """Sets the GitHub authentication details to be used by the GitHub api.
+    Args:
+        key (str): new username
```
```suggestion
        user (str): new username
```

tuxinal committed:
Correct documentation

Co-authored-by: osfanbuff63 <91388253+osfanbuff63@users.noreply.github.com>

osfanbuff63 reviewed (approved):
Looks good!

osfanbuff63 commented:
@all-contributors add @tuxinal for code
